NuGetResources/NuGetServiceInfo.cs

```diff
@@ -0,0 +1,103 @@
+﻿//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+namespace Microsoft.PackageManagement.NuGetProvider
+{
+    using Microsoft.PackageManagement.Provider.Utility;
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Parse and store a NuGet endpoint.
+    /// </summary>
+    public class NuGetServiceInfo
+    {
+        /// <summary>
+        /// Maps known @type values to preference attributes. New @type values should be added only after testing.
+        /// </summary>
+        private static readonly Dictionary<string, NuGetServiceInfo> KnownServiceTypeStrings = new Dictionary<string, NuGetServiceInfo>()
+        {
+            { "searchqueryservice", new NuGetServiceInfo(NuGetServiceType.Query, 0) },
+            { "searchqueryservice/3.0.0-beta", new NuGetServiceInfo(NuGetServiceType.Query, 1) },
+            { "searchqueryservice/3.0.0-rc", new NuGetServiceInfo(NuGetServiceType.Query, 2) },
+
```

Comment:
Since 3.0.0 search service is GA (RTM), why do we need to support beta and rc versions of search services?

Reply:
For example, NuGet.exe uses "SearchAutocompleteService/3.0.0-beta" - from this we can extrapolate that "searchautocompleteservice" < "searchautocompleteservice/3.0.0-beta"

---
In reply to: [154537556](https://github.com/OneGet/NuGetProvider/pull/25#discussion_r154537556) [](ancestors = 154537556)

Reply:
How is this exposed to the users of NuGet provider?